go/oasis-test-runner/scenario/e2e/runtime/runtime_upgrade.go

```diff
@@ -106,9 +109,80 @@ func (sc *runtimeUpgradeImpl) Run(ctx context.Context, childEnv *env.Env) error
 	// Run client again.
 	sc.Logger.Info("starting a second client to check if runtime works")
 	sc.Scenario.TestClient = NewTestClient().WithSeed("seed2").WithScenario(InsertRemoveEncWithSecretsScenarioV2)
+
+	// Ensure that after upgrade, every compute worker had its old exploded
+	// bundle removed from its bundles dir.
+	for _, worker := range sc.Net.ComputeWorkers() {
+		if err := sc.verifyBundleDir(ctx, worker); err != nil {
+			sc.Logger.Error("compute worker bundle dir clean-up error",
```

Comment:
This log could be omitted as you already log in `verifyBundleDir`, and the error will be logged anyway.